LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_ECATAutoRestart.TcPOU

```diff
@@ -0,0 +1,132 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_ECATAutoRestart" Id="{7eb4a159-5b39-429e-b4d6-be42c06ba58e}" SpecialFunc="None">
+    <Declaration><![CDATA[
+FUNCTION_BLOCK FB_ECATAutoRestart
+VAR_INPUT
+    sNetId     : T_AmsNetId;
+    nSlaveAddr : UINT;
+    bEnable 	: BOOL;
+END_VAR
+VAR_IN_OUT
+END_VAR
+VAR_OUTPUT
+    bError 		: BOOL;
+    nErrorId	: UDINT;
+END_VAR
+VAR
+    // FB
+    fbgetSlaveState : FB_EcGetSlaveState;
+    fbsetSlaveState : FB_EcSetSlaveState;
+    fbTON			: TON;
+    fbTP			: TP;
+    bStartTON		: BOOL := TRUE;
+    bStateMError	: BOOL;
+    nNextState		: WORD;
+
+    // Used by Statemachine method
+    bStopTrying		: BOOL := FALSE;
+    nErrorCount		: UINT := 0;
+    nErrorMax		: UINT := 5;
+    fbLogger		: FB_LogMessage := (eSubsystem:= E_Subsystem.FIELDBUS);
+
+
+
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[
+fbTON(IN:= bStartTON OR NOT fbTON.Q, PT:= T#1s);
+IF bStartTON THEN bStartTON := FALSE; END_IF
+fbgetSlaveState(sNetId:=sNetId, nSlaveAddr:=nSlaveAddr,  bExecute:=fbTON.Q);
+
+StateMachine(nCurrentState:= fbgetSlaveState.state.deviceState,
+            bEnable:= bEnable,
+            bSetError:= fbsetSlaveState.bError,
+            bSetBusy:= fbsetSlaveState.bBusy,
+            nNextState=> nNextState,
+            bMaxNumTriesReached=> bStateMError,
+            bSetDO=>fbTP.IN);
+
+bError := bStateMError;
+nErrorId := fbgetSlaveState.nErrId;
+
+fbTP(PT:=T#10ms);
```

Comment:
Can you explain the idea here with using a fixed-length pulse for the execute? Is this something special because of how the state setting works? Naiively I'd assume there's a way to do this that doesn't rely on specific hard-coded timespans.